### PR TITLE
Create calendar component in camp creation scheduled sessions

### PR DESCRIPTION
**Pull request by e-wai**
## Notion ticket link

[Camp Creation/Editing: Scheduling Sessions Calendar Component](https://www.notion.so/uwblueprintexecs/Task-Board-F22-9e03bf9f7c4343f09a8a5c16dc48a12a?p=97be9a874e95473786f511f136ac6dd6&pm=c)

## Implementation description
*  **Used https://fullcalendar.io/** (all devs will have to rebuild on merge)
*  Custom header component to have our styling for today button, and Chakra input selectors
* Moved calendar / side panel components into separate directories

## Steps to test
1. Create camp and add sessions with different dates, does this match what we expect? Test case with 8+ sessions to make sure colours loop properly
2. Check navigation functionality by going between months and years, select today

https://user-images.githubusercontent.com/40974372/202888747-e3d6ad6b-0178-443c-acfd-55ca319ac1b9.mov

## What should reviewers focus on?
* Some messy CSS handling + responsiveness that people may want to take a look at (should check out the 1980x1080px dimensions)

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/CampCreation/ScheduleSession/index.tsx
Comment (rchang0501) [inline frontend/src/components/pages/CampCreation/ScheduleSession/index.tsx:6]: Is `scheduledSessions` supposed to be a `Array<Array<string>>` or is this just for dummy data? Wouldn't it be better for this to be a `CampSession[]`?

**Comment by dhmi19** [comment]
Found a bug with the month + years drop downs not matching the calendar view when you click on the "Today" button. 

https://user-images.githubusercontent.com/57159970/202932043-083cdebd-3105-4aad-a0ea-d2e167c3830c.mov

File: frontend/src/components/pages/CampCreation/ScheduleSessions/SidePanel/SessionSidePanel.tsx
Comment (dhmi19) [inline frontend/src/components/pages/CampCreation/ScheduleSessions/SidePanel/SessionSidePanel.tsx:12]: why the React.FC instead of normal Element here?

File: frontend/src/components/pages/CampCreation/ScheduleSessions/SidePanel/index.tsx
Comment (dhmi19) [inline frontend/src/components/pages/CampCreation/ScheduleSessions/SidePanel/index.tsx:21]: not part of this ticket, but do you think we can make the cards for the sessions a little smaller? On my 13 inch mac, it looks like this:
image

I think 13 inch computers should be a common screen size?

File: frontend/src/components/pages/CampCreation/ScheduleSessions/SidePanel/ScheduledSessionsCard.tsx
Comment (dhmi19) [inline frontend/src/components/pages/CampCreation/ScheduleSessions/SidePanel/ScheduledSessionsCard.tsx:17]: is this branch rebased? I tried deleting sessions from the current sessions panel but it didn't show the delete session modal to confirm delete. I thought Ryan already changed that on his end?
image

File: frontend/src/constants/CampManagementConstants.ts
Comment (dhmi19) [inline frontend/src/constants/CampManagementConstants.ts:1]: could we clean up these comments if not needed anymore

File: frontend/src/constants/CampManagementConstants.ts
Comment (dhmi19) [inline frontend/src/constants/CampManagementConstants.ts:33]: if we hardcode the years, wouldn't the application technically break once we hit 2032 lol. Why not construct the years based on the current year?

**Review by dhmi19 — CHANGES_REQUESTED** [review]
functionality looks mostly good. Thanks for all the work on this!

File: frontend/src/components/pages/CampCreation/ScheduleSessions/SidePanel/ScheduledSessionsCard.tsx
Comment (e-wai) [reply frontend/src/components/pages/CampCreation/ScheduleSessions/SidePanel/ScheduledSessionsCard.tsx:17]: Hadn't rebased, will be merged in https://github.com/uwblueprint/focus-on-nature/pull/161

File: frontend/src/constants/CampManagementConstants.ts
Comment (e-wai) [reply frontend/src/constants/CampManagementConstants.ts:33]: Yeah, tbh I just assumed that was OK with us but will change to current year + 10

**Comment by e-wai** [comment]
New scrolling behaviour, size changes, colours on session card borders 

https://user-images.githubusercontent.com/40974372/204083783-534fecb6-18c8-471f-bad9-3f3ae213f438.mov

**Comment by e-wai** [comment]
There's one more issue that the cards are too big horizontally rn, but will fix in the morning (and also will open separate PR for linting repo, bunch of warnings lying around)

File: frontend/src/components/pages/CampCreation/ScheduleSessions/SidePanel/CurrentSessionsView.tsx
Comment (e-wai) [inline frontend/src/components/pages/CampCreation/ScheduleSessions/SidePanel/CurrentSessionsView.tsx:80]: https://stackoverflow.com/questions/70198644/overflow-scroll-on-div-with-flex-1-without-specific-height

**Comment by dhmi19** [comment]
image
Some design comments- not meant to be a blocker for merging. This is how the side panel looks like on my mac screen. Do we want to make the scrollable cards section a bit longer? It feels a little small right now. Also, the wrap is a little off (I am guessing due to the padding/margin. If we need to merge to move on, I think it's fine to revisit this or prioritize

**Review by dhmi19 — APPROVED** [review]
good to go (pending verdict on the design changes)

**Comment by e-wai** [comment]
@dhmi19 what size screen are you on?

re: longer scrollable container, I think it'll overlap on the footer if we push it any further. I'll hold off on extending it until the footer gets created/merged

**Comment by dhmi19** [comment]
> @dhmi19 what size screen are you on?
> 
> re: longer scrollable container, I think it'll overlap on the footer if we push it any further. I'll hold off on extending it until the footer gets created/merged

I am on a 13 inch mac. I was thinking of extending it to be inline with the bottom of the calendar component?

**Pull request merged**
